zpay32/decode.go

```diff
@@ -186,6 +187,13 @@ func Decode(invoice string, net *chaincfg.Params, opts ...DecodeOption) (
 			return nil, fmt.Errorf("unable to deserialize "+
 				"signature: %v", err)
 		}
+		// Ensure the signature is in canonical low-S form.
+		sValue := new(big.Int).SetBytes(sigBase256[32:64])
+		halfOrder := new(big.Int).Rsh(btcec.S256().N, 1)
+		if sValue.Cmp(halfOrder) > 0 {
```

Comment:
Perhaps let's get a helper function in `btcec`?

Reply:
We can extract it and re-use from here: https://github.com/btcsuite/btcd/blob/9ff0780da683ec1fa88ce9b9f2f4fd39d5592221/txscript/engine.go#L1399-L1412. 

Also the bounds on the slice above my not be entirely correct, as the signature encoding itself can be variable size.